### PR TITLE
DSD-1710: Use onClearAll  function in multi-select component

### PR DESCRIPTION
**Pull request by gonuguntla**
Fixes JIRA ticket xxx

## This PR does the following:

- Used onClearAll function in multiSelect component instead of onClear.
- Updated onClearAll function in multiSelectGroup and FilterBar.
- Fixed focus issue when click on the ViewAll button.
- Fixed scroll issue when isBlockElement flag is false.

## How has this been tested?
Local and storybook

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/theme/components/multiSelect.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ const MultiSelect = defineMultiStyleConfig({
       },
       ".chakra-collapse": {
         bgColor: "ui.bg.page",
+        overflow: "unset !important",
```
Comment (gonuguntla) [inline src/theme/components/multiSelect.ts:113]: This code is related to [DSD-1713](https://jira.nypl.org/browse/DSD-1713)

File: src/components/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -222,6 +224,11 @@ export const MultiSelect: ChakraComponent<
       /** Toggle for listOverflow = "expand" */
       const toggleItemsList = () => {
         setIsExpandable((prevProp) => !prevProp);
+        setTimeout(() => {
```
Comment (gonuguntla) [inline src/components/MultiSelect/MultiSelect.tsx:227]: This code is related to DSD-1712. I tried without setTimeout, but focus of the button is not working after onClick.

File: src/components/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -94,6 +94,8 @@ export const MultiSelect: ChakraComponent<
       // Create a ref to hold a reference to the accordian button, enabling us to programmatically focus it.
       const accordianButtonRef: React.RefObject<HTMLDivElement> =
         useRef<HTMLDivElement>();
+      const buttonRef: React.RefObject<HTMLButtonElement> =
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelect.tsx]: Rename this `expandToggleButtonRef` to be a bit more clear.

**Review by EdwinGuzman — APPROVED** [review]
The overflow issue seems like it'll work as well as the button focus behavior (just rename the variable).

The `onClearAll` seems like it will work. I thought you would need both `onClear` and `onClearAll` but that does not seem like that's the case. This _might_ change for `MultiSelectGroup` and/or `FilterBar` but we can update this later. I can't say I know why the `onClear` was  there to begin with if the `onChange` is doing the addition and removal in the `selectedItems` array.

**Review by EdwinGuzman — APPROVED** [review]

**Review by bigfishdesign13 — APPROVED** [review]

**Pull request merged**